Index: SHERPA/Single_Events/Signal_Processes.C
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/SHERPA/Single_Events/Signal_Processes.C b/SHERPA/Single_Events/Signal_Processes.C
--- a/SHERPA/Single_Events/Signal_Processes.C	(revision 46b8f9eacfafa9416be09411c059cdd30a9a1bcb)
+++ b/SHERPA/Single_Events/Signal_Processes.C	(date 1745593239979)
@@ -32,9 +32,9 @@
   p_yfshandler = mehandler->GetYFS();
   if (p_remnants[0]==NULL || p_remnants[1]==NULL)
     THROW(critical_error,"No beam remnant handler found.");
-  Scoped_Settings metssettings{
-    Settings::GetMainSettings()["METS"] };
-  m_cmode=metssettings["CLUSTER_MODE"].Get<int>();
+  Scoped_Settings mepssettings{
+    Settings::GetMainSettings()["MEPS"] };
+  m_cmode=mepssettings["CLUSTER_MODE"].Get<int>();
   Scoped_Settings spsettings{ Settings::GetMainSettings()["SP"] };
   m_setcolors = spsettings["SET_COLORS"].SetDefault(false).Get<bool>();
   m_adddocumentation = spsettings["ADD_DOC"].SetDefault(false).Get<bool>();

