diff -u -r MCFM/src/W1jet//A51.f ../MCFM/src/W1jet//A51.f
--- MCFM/src/W1jet//A51.f	2009-01-23 09:27:35.000000000 +0000
+++ ../MCFM/src/W1jet//A51.f	2011-01-24 14:26:29.000000000 +0000
@@ -5,6 +5,7 @@
       include 'sprods_com.f'
       include 'scale.f'
       include 'epinv.f'
+      include 'epinv2.f'
       integer j1,j2,j3,j4,j5
       double complex Vcc,Fcc,Vsc,Fsc,l12,l23,L0,L1,Lsm1,A5lom
       double complex lnrat
@@ -16,8 +17,8 @@
 
 C--leading N
       Vcc=
-     . -(epinv**2+epinv*l12+0.5d0*l12**2)
-     . -(epinv**2+epinv*l23+0.5d0*l23**2)
+     . -(epinv2+epinv*l12+0.5d0*l12**2)
+     . -(epinv2+epinv*l23+0.5d0*l23**2)
      . -2d0*(epinv+l23)-4d0      
 
       Fcc=za(j3,j4)**2/(za(j1,j2)*za(j2,j3)*za(j4,j5))
diff -u -r MCFM/src/W1jet//A52.f ../MCFM/src/W1jet//A52.f
--- MCFM/src/W1jet//A52.f	2009-01-23 09:27:35.000000000 +0000
+++ ../MCFM/src/W1jet//A52.f	2011-01-24 15:05:27.000000000 +0000
@@ -5,6 +5,7 @@
       include 'sprods_com.f'
       include 'scale.f'
       include 'epinv.f'
+      include 'epinv2.f'
       integer j1,j2,j3,j4,j5
       double complex Vcc,Fcc,Vsc,Fsc,l12,l45,L0,L1,Lsm1,A5lom
       double complex lnrat
@@ -13,7 +14,7 @@
       l45=lnrat(musq,-s(j4,j5))  
 C    -i * A5tree  
       A5lom=za(j2,j4)**2/(za(j2,j3)*za(j3,j1)*za(j4,j5))
-      Vcc=-(epinv**2+epinv*l12+0.5d0*l12**2)
+      Vcc=-(epinv2+epinv*l12+0.5d0*l12**2)
      . -2d0*(epinv+l45)-4d0
 
 C--subleading N
diff -u -r MCFM/src/W1jet//qqb_w1jet_v.f ../MCFM/src/W1jet//qqb_w1jet_v.f
--- MCFM/src/W1jet//qqb_w1jet_v.f	2009-01-23 09:27:35.000000000 +0000
+++ ../MCFM/src/W1jet//qqb_w1jet_v.f	2011-01-24 17:34:03.000000000 +0000
@@ -12,6 +12,7 @@
       include 'ckm.f'
       include 'sprods_com.f'
       include 'zprods_com.f'
+      include 'scale.f'
       include 'epinv.f'
       include 'scheme.f'
       include 'nflav.f'
